index.bs

```diff
@@ -2743,6 +2743,27 @@ and makes it clear when the state described by the flags is reset.
 
 <!-- TODO: add examples -->
 
+<h3 id="monkey-patching">Avoid monkey patching</h3>
+A <dfn export>monkey patch</dfn> layers new functionality on top of an existing specification in a way that extends, overrides, or otherwise modifies the underlying specification's behavior.
+Monkey patching <strong>is considered bad practice and should be avoided</strong> for the reasons listed below.
+
+An example of monkey patching would be: specification A defines an algorithm to do something, then specification B overrides or modifies that algorithm with some new functionality.
```

Comment:
```suggestion
An example of monkey patching would be: specification A defines an internal algorithm part of its capability, then specification B overrides or modifies said algorithm with functionality directly, not using publicly defined extension points.
```

Reply:
The abstract example is a bit confusing to the reader as it almost implies that no spec should extend another spec. My proposed change could be a good start to make the text a bit explicit. Ideally, we will have examples from the past to illustrate what a monkey patch is and why it is bad. We should also clarify the difference between monkey patching (bad) and modularization (good). Maybe also link to https://en.wikipedia.org/wiki/Monkey_patch